src/state/sagas/solution/FindSolutionById/FindSolutionByIdData.js

```diff
@@ -7,24 +7,41 @@ import { STATUSES } from '../../../commons/Constants';
 import { Api } from '../../../../services/config/Api';
 import { ConfigUtils, SolutionsUtils } from '../../../../utils';
 import { SolutionSchema } from '../../../../services/config/SolutionSchema';
+import { t } from 'i18next';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
 
 export function* fetchSolutionByIdData(organizationId, solutionId) {
-  const { data } = yield call(Api.Solutions.findSolutionById, organizationId, solutionId);
-  SolutionsUtils.castMinMaxDefaultValuesInSolution(data);
-  SolutionsUtils.patchSolutionIfLocalConfigExists(data);
-  ConfigUtils.checkUnknownKeysInConfig(SolutionSchema, data);
-  SolutionsUtils.checkParametersValidationConstraintsInSolution(data);
-  ConfigUtils.checkDeprecatedKeysInConfig(data);
-  SolutionsUtils.patchIncompatibleValuesInSolution(data);
+  try {
+    const { data } = yield call(Api.Solutions.findSolutionById, organizationId, solutionId);
+    SolutionsUtils.castMinMaxDefaultValuesInSolution(data);
+    SolutionsUtils.patchSolutionIfLocalConfigExists(data);
+    ConfigUtils.checkUnknownKeysInConfig(SolutionSchema, data);
+    SolutionsUtils.checkParametersValidationConstraintsInSolution(data);
+    ConfigUtils.checkDeprecatedKeysInConfig(data);
+    SolutionsUtils.patchIncompatibleValuesInSolution(data);
 
-  SolutionsUtils.addRunTemplatesParametersIdsDict(data);
-  SolutionsUtils.addTranslationLabels(data);
+    SolutionsUtils.addRunTemplatesParametersIdsDict(data);
+    SolutionsUtils.addTranslationLabels(data);
 
-  yield put({
-    type: SOLUTION_ACTIONS_KEY.SET_CURRENT_SOLUTION,
-    status: STATUSES.SUCCESS,
-    solution: data,
-  });
+    yield put({
+      type: SOLUTION_ACTIONS_KEY.SET_CURRENT_SOLUTION,
+      status: STATUSES.SUCCESS,
+      solution: data,
+    });
+  } catch (error) {
+    console.error(error);
+    yield put(
+      dispatchSetApplicationErrorMessage(
+        error,
+        t('commoncomponents.banner.', "You don't have permission to access this solution.")
```

Comment:
The translated key is invalid: `commoncomponents.banner.`
Also, we should probably add the solution id in the error message

Reply:
It was a temporary empty key, while fixing the main problem.